x-pack/solutions/observability/plugins/observability_agent/server/tools/get_change_points/get_metric_change_points.ts

```diff
@@ -0,0 +1,334 @@
+    ),
+  end: z
+    .string()
+    .describe(
+      'The end of the time range, in Elasticsearch datemath, like `now`, or an ISO timestamp'
+    ),
+  metrics: z
+    .array(
+      z.object({
+        name: z.string().describe('The name of the set of metrics'),
+        index: z.string().describe('The index or index pattern to find the metrics').optional(),
+        kqlFilter: z
+          .string()
+          .describe('A KQL filter to filter the metric documents, e.g.: my_field:foo')
+          .optional(),
+        field: z
+          .string()
+          .describe(
+            'Metric field that contains the metric. Only use if the metric aggregation type is not `count`.'
+          )
+          .optional(),
+        type: z
+          .enum(['count', 'avg', 'sum', 'min', 'max', 'p95', 'p99'])
+          .describe('The type of metric aggregation to perform. Defaults to `count`')
+          .optional(),
+        groupBy: z
+          .array(z.string())
+          .describe('Optional keyword fields to group metrics by.')
+          .optional(),
+      })
+    )
+    .describe(
+      'Analyze changes in metrics. DO NOT UNDER ANY CIRCUMSTANCES use date or metric fields for groupBy. Leave empty unless needed.'
+    ),
+});
+
+export function createObservabilityGetMetricChangePointsTool({
+  core,
+  plugins,
+  logger,
+}: {
+  core: CoreSetup<ObservabilityAgentPluginStartDependencies, ObservabilityAgentPluginStart>;
+  plugins: ObservabilityAgentPluginSetupDependencies;
+  logger: Logger;
+}) {
+  const toolDefinition: BuiltinToolDefinition<typeof getMetricChangePointsSchema> = {
+    id: OBSERVABILITY_GET_METRIC_CHANGE_POINTS_TOOL_ID,
+    type: ToolType.builtin,
+    description: 'Return change points such as spikes and dips for metrics.',
+    schema: getMetricChangePointsSchema,
+    tags: ['observability', 'metrics'],
+    handler: async ({ start, end, metrics = [] }, { esClient }) => {
+      try {
+        if (metrics.length === 0) {
+          throw new Error('No metrics found');
+        }
+
+        const metricIndexPatterns = await getMetricsIndices({ core, plugins, logger });
+
+        const metricChangePoints = await Promise.all([
+          ...metrics.map(async (metric) => {
+            const changePoints = await getMetricChangePoints({
+              index: metric.index || metricIndexPatterns.join(','),
+              esClient,
+              filters: getFilters({ start, end, kqlFilter: metric.kqlFilter }),
+              groupBy: metric.groupBy ?? [],
+              type: metric.type || 'count',
+              field: metric.field,
+            });
+
+            return changePoints.map((change) => ({
+              name: metric.name,
+              ...change,
+            }));
+          }),
+        ]);
+
+        const allMetricChangePoints = orderBy(metricChangePoints.flat(), [
+          (item) => ('p_value' in item.changes ? item.changes.p_value : Number.POSITIVE_INFINITY),
```

Comment:
You control the output from `getMetricChangePoints`. Make `p_value` required instead of this.